code-challenges.test.js

```diff
@@ -65,3 +133,21 @@ const accountTransactions3 = []
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+
+// GREEN
+// declare a function (arraySum) that takes in an array of numbers and returns the accumulated sum (meaning that every addition is logged, rather than the total sum of the numbers)
+// use the .map() method to iterate over the array and return a new array of numbers after following a provided action
+// set sum to initiate at 0 so that it does not add anything to the value at the first index
+// input: an array of numbers
+// output: a new array of numbers, showing the cumulative sum of the original array
+
+const arraySum = arr => arr.map((sum => value => sum += value)(0))
+
+console.log((arraySum(accountTransactions1)))
+// [100, 83, 60, 51]
+console.log((arraySum(accountTransactions2)))
+// [250, 161, 261, 165]
+console.log((arraySum(accountTransactions3)))
+// []
+
+// Note: My code was inspired heavily by a tutorial on the internet I found on how to find the cumulative sum of numbers in an array. I was really struggling to figure out how and why this code should work, but after finding information on the web, I understand way more. This code uses the .map() method, but rather than breaking it into multiple lines of code, it is simplified to one. This forgoes the need for curly brackets or a return, and now makes much more sense to me than it did when I was trying to write it as a traditional arrow function.
```

Comment:
I love that you found this and immediately ran to find sources and documentation to fully understand it before implementing it into your code. In going down this rabbit hole you now have another tool in the box to help your future code. This is a great and noble trait for any developer to have. That is very humble, Moss.